examples/finding_best_open_source_embedding_model/best_embedding_model_rag_app.ipynb

```diff
@@ -0,0 +1,559 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we explore the process of selecting the most suitable open-source embedding model for your Retrieval-Augmented Generation (RAG) application. \n",
+    "\n",
+    "We evaluate three open-source embedding models: [nomic-embed-text](https://ollama.com/library/nomic-embed-text), [bge-m3](https://ollama.com/library/bge-m3/blobs/daec91ffb5dd), and [mxbai-embed-large](https://ollama.com/library/mxbai-embed-large/blobs/819c2adf5ce6).\n",
+    " \n",
+    "To facilitate the execution of these models and the generation of embeddings, we will use the following tools:\n",
+    "\n",
+    "- [Ollama](https://ollama.com/), a platform that provides access to a variety of open-source Large Language Models (LLMs).\n",
+    "- [pgai](https://github.com/timescale/pgai), an open-source extension that seamlessly integrates LLM workflows, such as embedding creation and management, directly into your PostgreSQL database.\n",
+    "\n",
+    "The evaluation process involves:\n",
+    "\n",
+    "1. Setting up a test environment with Ollama and PostgreSQL\n",
+    "2. Loading Paul Graham's essays as our test dataset\n",
+    "3. Generating embeddings using different models\n",
+    "4. Creating diverse test questions across multiple categories\n",
+    "5. Evaluating each model's retrieval performance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Environment Setup\n",
+    "\n",
+    "Before you begin, make sure to update the `OPENAI-API-KEY` in the `compose.yaml` file. Then execute the `install.sh` shell script to set up your environment. This script will:\n",
+    "\n",
+    "1. Install all necessary Docker containers required for the environment.\n",
+    "2. Load the dataset into the PostgreSQL Docker container, which will be used throughout this tutorial.\n",
+    "\n",
+    "**Note for macOS Users:**\n",
+    "  \n",
+    "To run the script, first make it executable by running the command: `chmod +x install.sh`. Then, execute the script with: `./install.sh`.\n",
+    "\n",
+    "Let's install the necessary Python libraries for this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install pandas python-dotenv psycopg2-binary Jinja2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset Ingestion\n",
+    "\n",
+    "Let's setup the PostgreSQL database and install [pgai](https://github.com/timescale/pgai)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os \n",
+    "import psycopg2\n",
+    "from dotenv import load_dotenv\n",
+    "\n",
+    "load_dotenv()\n",
+    "\n",
+    "DATABASE_CONNECTION_STRING = os.environ[\"DATABASE_CONNECTION_STRING\"]\n",
```

Comment:
I got stuck here because this env variable wasn't set.

Reply:
I decided to remove the requirement to setup the environment variables instead I explicitly set them in the notebook.